third_party/move/move-vm/runtime/src/storage/script_storage.rs

```diff
@@ -25,13 +26,7 @@ pub trait ScriptStorage {
         serialized_script: &[u8],
     ) -> PartialVMResult<Arc<CompiledScript>>;
 
-    /// Returns a verified script, if it is cached. If not, the script is created using
-    /// the passed callback function. It is the responsibility of the client to ensure
-    /// that the callback verifies the script. An error is returned if script fails to
-    /// deserialize or verify.
-    fn fetch_or_create_verified_script(
-        &self,
-        serialized_script: &[u8],
-        f: &dyn Fn(Arc<CompiledScript>) -> PartialVMResult<Script>,
-    ) -> PartialVMResult<Arc<Script>>;
+    /// Returns a verified script. The script can either be cached, or deserialized and/or
+    /// verified from scratch. An error is returned if script fails to deserialize or verify.
+    fn fetch_verified_script(&self, serialized_script: &[u8]) -> PartialVMResult<Arc<Script>>;
```

Comment:
Scripts are not stored persistently, ScriptStorage is just a temporary storage for a higher-level representation of scripts, and there is no way for it to resolve the script by itself. That's why "fetch_xxx" seems off to me as the &[u8] has to always be provided. Just verify_script & deserialize_script would feel slightly better to me (or ..._with_cache or smt if we want to make that part clear?)

Reply:
I was not thinking about this from that perspective 😄 I like it! Again, decided to move it to #14402. Changed to `deserialize_and_cache_script` and `verify_and_cache_script`, what do you think?

Reply:
I like it much better 👍